Fix failing tests and documentation

vhirtham opened the pull request:
## Changes

- Fix failing tests due to new pandas version ->seems like pandas added own `__add__` functions so that we couldn't use `+` with a pandas Type on the left-hand side and our `Time` class on the right-hand side
- Fix failing documentation due to a cross reference -> `weldx.time.Time` vs `weldx.Time`

## Related Issues

None

## Checks

- [x] updated CHANGELOG.rst
- [x] updated tests
- [x] updated doc/
- [x] update example/tutorial notebooks
- [x] ~~update manifest file~~

vhirtham commented:
Finally, I got the documentation bug fixed... sort of ([see here](https://github.com/BAMWelDX/weldx/pull/684/commits/2b2bcf8b42dd2e98ea23b668bd9249191689d139)). During the process I went nuts and adjusted all imports to import from the correct module instead of the global weldx namespace. Maybe we have to review where we want to undo that (doc examples, tutorials?). On the plus side, it should probably also fix some circular imports that poped up every now and then.

marscher commented:
> 
Thanks for fixing this pandas issue.

Was it really mandatory to change all the imports? What was the exact error message?

vhirtham commented:
> > 
> 
> Thanks for fixing this pandas issue.
> 
> Was it really mandatory to change all the imports? What was the exact error message?

The fix for the pandas issue was the first commit and I just had to adjust the tests to the new behaviour.

The fix for the documantation was the linked commit in my previous comment. So the import changes weren't mandatory. However, I couldn't reproduce the bug locally so I had to try it by committing over and over again. I initially thaught sphinx was getting the `weldx.Time` definition from an direct weldx import. So I changed them to import from the correct module. After I changed all I could find, the error was still not fixed, so I removed the Type hint entirely and that did the trick.

As I said, we can undo all the import changes, but I think at least some of them do make sense since I get circular import problems from time to time when I run the debugger. But in the notebooks and examples, I feel we should undo it since it bloats the code and gives the wrong impression that you always have to pick the correct module. Feel free to mark things to be undone. Maybe we can discuss this during todays jour fixe.

This was the sphinx error message:

~~~
/home/runner/work/weldx/weldx/weldx/geometry.py:docstring of weldx.geometry.SpatialData.time:: WARNING: more than one target found for cross-reference 'Time': weldx.time.Time, weldx.Time
~~~

CagtayFabry commented on weldx/core.py:
```diff
@@ -909,7 +909,8 @@ def __init__(
         starting at point ``[0, 2 ,2] cm``
 
-        >>> from weldx import GenericSeries, Q_
+        >>> from weldx.core import GenericSeries
+        >>> from weldx.constants import Q_
```
is there a bug/import error when we import from the main `weldx` namespace in the doctests?

vhirtham replied:
See my comment above, no there isn't.

CagtayFabry commented:
Ok I just now caught up on the comments

I think in the python package files it is better/safer to import from the correct namespace to avoid potential import errors.
In the doctest examples that should not be needed and makes it less readable, so my vote goes to revert to top level imports here

Regarding the pandas issues:
From what I see we now skip the pandas types in the tests, but what is the actual behavior/error that comes up?
Can you maybe link to a CI run that failed with the new pandas version @vhirtham ?

CagtayFabry commented:
regarding the sphinx error message `WARNING: more than one target found for cross-reference 'Time': weldx.time.Time, weldx.Time`

I think this is mostly due to the fact that we have both elements `weldx.Time` and `weldx.time.Time` in our documentation toctree after #437 ?

marscher commented:
> regarding the sphinx error message `WARNING: more than one target found for cross-reference 'Time': weldx.time.Time, weldx.Time`
> 
> I think this is mostly due to the fact that we have both elements `weldx.Time` and `weldx.time.Time` in our documentation toctree after #437 ? So nothing related to the python package per se?

The same thing applies to CSM and LCS for instance, since they live in their own modules and are pulled to the top-level namespace. So there was a trick, how to avoid this ambiguity, but I cannot wrap my head around it right now.

vhirtham commented:
> Regarding the pandas issues: From what I see we now skip the pandas types in the tests, but what is the actual behavior/error that comes up? Can you maybe link to a CI run that failed with the new pandas version @vhirtham ?

Here you go: https://github.com/BAMWelDX/weldx/runs/4936771641?check_suite_focus=true

CagtayFabry commented:
> > Regarding the pandas issues: From what I see we now skip the pandas types in the tests, but what is the actual behavior/error that comes up? Can you maybe link to a CI run that failed with the new pandas version @vhirtham ?
> 
> Here you go: https://github.com/BAMWelDX/weldx/runs/4936771641?check_suite_focus=true

Interesting, it looks like `TimedeltaIndex + Time` now returns a `TimedeltaIndex` even though I don't know how they do it... (I guess because we implemented `__iter__` or similar so `Time` is accessed as an array)

CagtayFabry commented:
I have merged the changes to fix the errors in the other PRs #689 and #678 @vhirtham 
We can emrge the internal import cleanups here

vhirtham commented on tutorials/01_02_time_dependent_data.ipynb:
```diff
@@ -347,9 +347,9 @@
     "What remains now is to divide the result by the process duration.\n",
     "We can get the time data either from the voltage or the current data as they were both measured simultaneously.\n",
-    "It can be accessed with `.time` which returns a `weldx.Time` instance.\n",
+    "It can be accessed with `.time` which returns a `weldx.time.Time` instance.\n",
```
```suggestion
    "It can be accessed with `.time` which returns a `weldx.Time` instance.\n",
```

vhirtham commented on tutorials/01_02_time_dependent_data.ipynb:
```diff
@@ -347,9 +347,9 @@
+    "It can be accessed with `.time` which returns a `weldx.time.Time` instance.\n",
     "We will talk a bit more about this class in another tutorial.\n",
-    "For now it is sufficient to know, that it provides a `.duration` method to get the duration of the contained time data as a new `weldx.Time` instance and that we can turn it into a quantity using `.as_quantity`.\n",
+    "For now it is sufficient to know, that it provides a `.duration` method to get the duration of the contained time data as a new `weldx.time.Time` instance and that we can turn it into a quantity using `.as_quantity`.\n",
```
```suggestion
    "For now it is sufficient to know, that it provides a `.duration` method to get the duration of the contained time data as a new `weldx.Time` instance and that we can turn it into a quantity using `.as_quantity`.\n",
```

vhirtham commented on weldx/constants.py:
```diff
@@ -43,7 +43,7 @@
 Examples
 --------
->>> from weldx import Q_
+>>> from weldx.constants import Q_
```
```suggestion
>>> from weldx import Q_
```

vhirtham commented on weldx/core.py:
```diff
@@ -909,7 +909,8 @@ def __init__(
         starting at point ``[0, 2 ,2] cm``
 
-        >>> from weldx import GenericSeries, Q_
+        >>> from weldx.core import GenericSeries
+        >>> from weldx.constants import Q_
```
```suggestion
        >>> from weldx import GenericSeries, Q_
```

vhirtham commented on weldx/measurement.py:
```diff
@@ -238,7 +238,7 @@ def __init__(
         Examples
         --------
-        >>> from weldx import Q_
+        >>> from weldx.constants import Q_
```
```suggestion
        >>> from weldx import Q_
```

vhirtham commented on weldx/measurement.py:
```diff
@@ -324,7 +324,7 @@ def from_equipment(
         Examples
         --------
-        >>> from weldx import Q_
+        >>> from weldx.constants import Q_
```
```suggestion
        >>> from weldx import Q_
```

vhirtham commented on weldx/measurement.py:
```diff
@@ -399,7 +399,7 @@ def from_parameters(
         Examples
         --------
-        >>> from weldx import Q_
+        >>> from weldx.constants import Q_
```
```suggestion
        >>> from weldx import Q_
```

vhirtham commented on weldx/measurement.py:
```diff
@@ -613,7 +613,7 @@ def add_transformation(
         Examples
         --------
-        >>> from weldx import Q_
+        >>> from weldx.constants import Q_
```
```suggestion
        >>> from weldx import Q_
```

vhirtham commented on weldx/measurement.py:
```diff
@@ -689,7 +689,7 @@ def add_transformation_from_equipment(
         Examples
         --------
-        >>> from weldx import Q_
+        >>> from weldx.constants import Q_
```
```suggestion
        >>> from weldx import Q_
```

vhirtham commented on weldx/measurement.py:
```diff
@@ -789,7 +789,7 @@ def create_transformation(
         Examples
         --------
-        >>> from weldx import Q_
+        >>> from weldx.constants import Q_
```
```suggestion
        >>> from weldx import Q_
```

vhirtham commented on weldx/time.py:
```diff
@@ -99,7 +99,8 @@ class 'TimeDependent' are supported too.
     Creation from a quantity:
 
-    >>> from weldx import Q_, Time
+    >>> from weldx.constants import Q_
+    >>> from weldx.time import Time
```
```suggestion
    >>> from weldx.constants import Q_, Time
```

vhirtham commented on weldx/time.py:
```diff
@@ -148,11 +149,11 @@ class 'TimeDependent' are supported too.
     passed directly to `Time` as `time` parameter:
 
-    >>> from weldx import LocalCoordinateSystem as LCS
+    >>> from weldx.transformations.local_cs import LocalCoordinateSystem as LCS
```
```suggestion
    >>> from weldx import LocalCoordinateSystem as LCS
```

vhirtham commented on weldx/time.py:
```diff
@@ -148,11 +149,11 @@ class 'TimeDependent' are supported too.
     >>> t_from_lcs = Time(lcs)
     >>>
-    >>> from weldx import TimeSeries
+    >>> from weldx.core import TimeSeries
```
```suggestion
    >>> from weldx import TimeSeries
```

vhirtham commented on weldx/time.py:
```diff
@@ -602,7 +603,7 @@ def resample(self, number_or_interval: Union[int, types_timedelta_like]):
         Examples
         --------
-        >>> from weldx import Time
+        >>> from weldx.time import Time
```
```suggestion
        >>> from weldx import Time
```

vhirtham commented on weldx/time.py:
```diff
@@ -739,7 +740,7 @@ def __get__(self, ins, typ):
     Using ``union`` as class method:
 
-    >>> from weldx import Time
+    >>> from weldx.time import Time
```
```suggestion
    >>> from weldx import Time
```

vhirtham commented on weldx/transformations/cs_manager.py:
```diff
@@ -785,16 +775,15 @@ def create_cs_from_axis_vectors(
         Create a coordinate system from 3 orthogonal vectors:
 
-        >>> from weldx import CoordinateSystemManager
+        >>> from weldx.transformations.cs_manager import CoordinateSystemManager
```
```suggestion
        >>> from weldx import CoordinateSystemManager
```

vhirtham commented on weldx/transformations/local_cs.py:
```diff
@@ -486,7 +486,7 @@ def from_axis_vectors(
         Create a coordinate system from 3 orthogonal vectors:
 
-        >>> from weldx import LocalCoordinateSystem
+        >>> from weldx.transformations.local_cs import LocalCoordinateSystem
```
```suggestion
        >>> from weldx import LocalCoordinateSystem
```

vhirtham commented:
@CagtayFabry All the doc changes are undone and some additional corrections applied. If you find the time to revisit, that would be nice. Otherwise, I ll merge tomorrow
